File: app/src/main/java/com/baidu/tv/player/model/DeviceCodeResponse.java
package com.baidu.tv.player.model;

import com.google.gson.annotations.SerializedName;

/**
 * 设备码响应模型
 */
public class DeviceCodeResponse {
    
    @SerializedName("device_code")
    private String deviceCode;
    
    @SerializedName("user_code")
    private String userCode;
    
    @SerializedName("verification_url")
    private String verificationUrl;
    
    @SerializedName("expires_in")
    private long expiresIn;
    
    @SerializedName("interval")
    private int interval;
    
    @SerializedName("error")
    private String error;
    
    @SerializedName("error_description")
    private String errorDescription;

    // Getters and Setters
    public String getDeviceCode() {
        return deviceCode;
    }

    public void setDeviceCode(String deviceCode) {
        this.deviceCode = deviceCode;
    }

    public String getUserCode() {
        return userCode;
    }

    public void setUserCode(String userCode) {
        this.userCode = userCode;
    }

    public String getVerificationUrl() {
        return verificationUrl;
    }

    public void setVerificationUrl(String verificationUrl) {
        this.verificationUrl = verificationUrl;
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(long expiresIn) {
        this.expiresIn = expiresIn;
    }

    public int getInterval() {
        return interval;
    }

    public void setInterval(int interval) {
        this.interval = interval;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getErrorDescription() {
        return errorDescription;
    }

    public void setErrorDescription(String errorDescription) {
        this.errorDescription = errorDescription;
    }

    /**
     * 是否有错误
     */
    public boolean hasError() {
        return error != null && !error.isEmpty();
    }
    
    /**
     * 获取完整的验证URL
     */
    public String getFullVerificationUrl() {
        if (verificationUrl == null || userCode == null) {
            return null;
        }
        return verificationUrl + "?code=" + userCode;
    }
}

File: app/src/main/java/com/baidu/tv/player/kt/model/DeviceCodeResponse.kt
package com.baidu.tv.player.kt.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * 设备码响应模型（百度 OAuth 设备码授权）。
 */
@Parcelize
data class DeviceCodeResponse(
    @SerializedName("device_code") val deviceCode: String? = null,
    @SerializedName("user_code") val userCode: String? = null,
    @SerializedName("verification_url") val verificationUrl: String? = null,
    @SerializedName("expires_in") val expiresIn: Long = 0,
    @SerializedName("interval") val interval: Int = 0,
    @SerializedName("error") val error: String? = null,
    @SerializedName("error_description") val errorDescription: String? = null,
) : Parcelable {
    /** 是否有错误。 */
    fun hasError(): Boolean = !error.isNullOrEmpty()

    /** 拼接带 user_code 的完整验证 URL（供 TV 端展示二维码）。 */
    val fullVerificationUrl: String?
        get() = if (verificationUrl == null || userCode == null) null
        else "$verificationUrl?code=$userCode"
}
